benchmarks/src/jmh/java/io/opencensus/benchmarks/trace/BasicOperationsBenchmark.java

```diff
@@ -0,0 +1,305 @@
+    span.end();
+    return span;
+  }
+
+  /** Create a child span. */
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public Span createSpanWithExplicitParent(Data data) {
+    Span span =
+        data.tracer
+            .spanBuilderWithExplicitParent("ChildSpan", data.span)
+            .setRecordEvents(data.recorded)
+            .setSampler(data.sampled ? Samplers.alwaysSample() : Samplers.neverSample())
+            .startSpan();
+    span.end();
+    return span;
+  }
+
+  /** Create a child span with a remote parent. */
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public Span createSpanWithRemoteParent(Data data) {
+    Span span =
+        data.tracer
+            .spanBuilderWithRemoteParent("ChildSpanFromRemoteParent", data.span.getContext())
+            .setRecordEvents(data.recorded)
+            .setSampler(data.sampled ? Samplers.alwaysSample() : Samplers.neverSample())
+            .startSpan();
+    span.end();
+    return span;
+  }
+
+  /** Create a child span from the current span. */
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public Span createSpanWithCurrentSpan(Data data) {
+    Span span =
+        data.tracer
+            .spanBuilder("ChildSpanFromCurrent")
+            .setRecordEvents(data.recorded)
+            .setSampler(data.sampled ? Samplers.alwaysSample() : Samplers.neverSample())
+            .startSpan();
+    span.end();
+    return span;
+  }
+
+  /** Create a link */
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public Link createLink(Data data) {
+    return Link.fromSpanContext(
+        SpanContext.create(
+            TraceId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 0}),
+            SpanId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 0}),
+            TraceOptions.DEFAULT,
+            TRACESTATE_DEFAULT),
+        Link.Type.PARENT_LINKED_SPAN);
+  }
+
+  /** Create a message event */
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public MessageEvent createMessageEvent(Data data) {
+    return MessageEvent.builder(MessageEvent.Type.SENT, MESSAGE_ID).build();
+  }
+
+  /** Scope/Unscope a trace span. */
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public Scope scopeSpan(Data data) {
+    try (Scope scope = data.tracer.withSpan(data.spanToScope)) {
+      return scope;
+    }
+  }
+
+  /** Get current trace span. */
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public Span getCurrentSpan(Data data) {
+    return data.tracer.getCurrentSpan();
+  }
+
+  /** Encode a span using binary format. */
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public byte[] encodeSpanBinary(Data data) {
```

Comment:
encode and decode will vary based on number of attributes, links and annotations. Will that be covered later?

Reply:
Encoding/Decoding only apply to the span context - tracestate will get added eventually but we don't support that at the moment.